### PR TITLE
feat: flatten contracts

### PR DESCRIPTION
**Pull request by iamdefinitelyahuman**
### What I did
Add functionality for flattening a contract.

### CODE REVIEW

**Review by Macarse — APPROVED** [review]

File: brownie/network/contract.py [hunk 1]
```diff
@@ -127,6 +127,32 @@ def decode_input(self, calldata: Union[str, bytes]) -> Tuple[str, Any]:
...
+                build_json = self._project._build.get(name)
+                offset = slice(*build_json["offset"])
+                source = build_json["source"][offset]
+                flattened_source = f"{flattened_source}\n\n{source}"
+
+            build_json = self._build
+            version = build_json["compiler"]["version"]
+            offset = slice(*build_json["offset"])
+            source = build_json["source"][offset]
+            flattened_source = f"pragma solidity {version};{flattened_source}\n\n{source}\n"
```
Comment (fubuloubu) [inline brownie/network/contract.py:150]:
```suggestion
            flattened_source = f"pragma solidity {version};\n\n{flattened_source}\n\n{source}\n"
```

Comment (iamdefinitelyahuman) [reply brownie/network/contract.py:150]: It's handled in line 144

**Comment by Macarse** [comment]
This can be tested without deploying a new version.
Open a brownie terminal and use the contract like this:

```
for name in my_contract._build["dependencies"]:
  build_json = my_contract._project._build.get(name)
  offset = slice(*build_json["offset"])
  source = build_json["source"][offset]
  flattened_source = f"{flattened_source}\n\n{source}"

build_json = my_contract._build
version = build_json["compiler"]["version"]
offset = slice(*build_json["offset"])
source = build_json["source"][offset]
flattened_source = f"pragma solidity {version};{flattened_source}\n\n{source}\n"

f = open( 'output.sol', 'w' )
f.write(flattened_source)
f.close()
```

**Comment by Macarse** [comment]
@iamdefinitelyahuman you force pushed a rebase?

**Comment by matnad** [comment]
Commit included in #914. Closing this.

**Pull request closed without merging**